paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -694,39 +699,35 @@ def autoscaling_is_paused():
 
 
 @use_requests_cache('service_autoscaler')
-def autoscale_services(soa_dir=DEFAULT_SOA_DIR):
+def autoscale_services(soa_dir=DEFAULT_SOA_DIR, services=None):
     if autoscaling_is_paused():
         log.warning("Skipping autoscaling because autoscaler paused")
         return
 
-    try:
-        with create_autoscaling_lock():
-            system_paasta_config = load_system_paasta_config()
-            cluster = system_paasta_config.get_cluster()
-            configs = get_configs_of_services_to_scale(cluster=cluster, soa_dir=soa_dir)
-
-            marathon_clients = get_marathon_clients(get_marathon_servers(system_paasta_config))
-            apps_with_clients = get_marathon_apps_with_clients(marathon_clients.get_all_clients(), embed_tasks=True)
-            all_mesos_tasks = a_sync.block(get_all_running_tasks)
-            if configs:
-                with ZookeeperPool():
-                    for config in configs:
-                        try:
-                            marathon_tasks, mesos_tasks = filter_autoscaling_tasks(
-                                [app for (app, client) in apps_with_clients],
-                                all_mesos_tasks,
-                                config,
-                            )
-                            autoscale_marathon_instance(
-                                config,
-                                system_paasta_config,
-                                list(marathon_tasks.values()),
-                                mesos_tasks,
-                            )
-                        except Exception as e:
-                            write_to_log(config=config, line='Caught Exception %s' % e)
-    except LockHeldException:
-        log.warning("Skipping autoscaling run for services because the lock is held")
+    system_paasta_config = load_system_paasta_config()
+    cluster = system_paasta_config.get_cluster()
+    configs = get_configs_of_services_to_scale(cluster=cluster, soa_dir=soa_dir, services=services)
+
+    marathon_clients = get_marathon_clients(get_marathon_servers(system_paasta_config))
+    apps_with_clients = get_marathon_apps_with_clients(marathon_clients.get_all_clients(), embed_tasks=True)
+    all_mesos_tasks = a_sync.block(get_all_running_tasks)
+    if configs:
```

Comment:
Do we need to nest here? Can't configs just be `[]` if there isn't anything to autoscale?

Reply:
I guess it stops us creating a ZK connection if we have nothing to do. But that is cheap so I guess I'll remove it